assets/src/edit-story/app/api/apiProvider.js

```diff
@@ -240,9 +240,46 @@ function APIProvider({ children }) {
     return apiFetch({ path });
   }, [statuses]);
 
-  const getAllUsers = useCallback(() => {
-    return apiFetch({ path: users });
-  }, [users]);
+  const getUsersByPage = useCallback(
+    (pagingNum = 1) => {
+      let apiPath = users;
+      const perPage = 100;
+      apiPath = addQueryArgs(apiPath, {
+        per_page: perPage,
+        page: pagingNum,
+      });
+      return apiFetch({ path: apiPath, parse: false }).then(
+        async (response) => {
+          const jsonArray = await response.json();
+
+          return { data: jsonArray, headers: response.headers };
+        }
+      );
+    },
+    [users]
+  );
+
+  const getAllUsers = useCallback(
+    (pagingNum = 1) => {
+      return getUsersByPage(pagingNum).then(({ data, headers }) => {
+        const totalPages = parseInt(headers.get('X-WP-TotalPages'));
+        if (totalPages <= 1) return data;
+        const promises = [];
+        for (let i = 2; i <= totalPages; i++) {
+          promises.push(getUsersByPage(i));
+        }
+        return Promise.all(promises).then((response) => {
+          return data.concat(
+            response.reduce(
+              (resultArray, result) => resultArray.concat(result.data),
+              []
+            )
+          );
+        });
+      });
```

Comment:
Don't we need to order the array now. What if page 5 return before page 4? This promises, you don't know what order they will be fulfilled

Reply:
hmm... Good point. Let me update it! 👍

Reply:
It should be sorted name field in the api.

Reply:
We need sort param as well? Or you mean it supposed to be sorted by name originally?

Reply:
Oh, it's original feature!

Reply:
I don't think it needs a sort param for now. I think before alpha we should change this to an autocomplete box. But for now, let's just go with sorting by name.

Reply:
Done!